applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -422,6 +440,9 @@ def _GetDistanceModificationProcess(self):
             self._distance_modification_process = self.__CreateDistanceModificationProcess()
         return self._distance_modification_process
 
+    def _CreateCustomizedLinearSolver(self, linear_solver_settings):
+        return linear_solver_factory.ConstructSolver(linear_solver_settings)
+
```

Comment:
Why you need this function? Calling the `_CreateLinearSolver` in the base class must work. In the derived MPI class you need to do nothing as the override is already there.

Reply:
In a near future, we will be obliged to pass specific settings for each solver (the conditions of the LSEs are different). 
Since we are now modifying the main solver, let's include the provisions.

Reply:
If this is the case, I'd modify the `_CreateLinearSolver()` method in the base `FluidSolver`class to accept a string, which takes as default `"linear_solver_settings"` rather than creating a new method. If no provided (as we do in the majority of cases) it will keep the current behavior. If provided it will create the solver with these.

As this change affects all the application solvers, needs to be done in a standalone PR. I strongly suggest using the current base class method until we modify the base class.

Reply:
Indeed you don't even need to override the Trilinos method if you use the same alias when importing the factory.

Reply:
Most probably the linear solver settings should be encapsulated within "levelset_convection_settings", "distance_reinitialization_settings", and "distance_smoothing_settings". So, I think the separate creator might be a better idea than modifying the fluid_solver.

Reply:
My design is precisely intended to support this... If not provided, the default argument will take `linear_solver_settings`. If provided, it will take wathever the user defines.

Reply:
So, as I understand, we will later modify the main function to take whatever setting that is provided (with the basic linear solver settings as the default). Ok. I will revert some changes.

Reply:
Exactly. :wink: